Library/Homebrew/formula.rb

```diff
@@ -166,6 +166,10 @@ def initialize(name, path, spec)
     validate_attributes!
     @build = active_spec.build
     @pin = FormulaPin.new(self)
+
+    if head && head_pkg_version = latest_head_pkg_version
```

Comment:
No, this checks whether `latest_head_pkg_version` is `nil` or not i.e. there is at least one  `HEAD` installed.